### PR TITLE
Alert dismiss callback

### PR DESCRIPTION
**Pull request by philschanely**
## Description

This PR adds ability to respond externally when the dismiss button within Sage Alert is clicked.

- Rails version now issues  a custom event `sage.alert.dismiss`
- React allows a callback function to provided through `onClickDismiss`

## Testing in `sage-lib`

See http://localhost:4000/pages/component/alert
See http://localhost:4100/?path=/docs/sage-alert--dismissable-alert
See http://localhost:4110/?path=/docs/sage-alert--dismissable-alert

## Testing in `kajabi-products`

1. (**LOW**) Sage Alert now allows for responding to clicks on dismiss button.

## Related

[SAGE-655](https://kajabi.atlassian.net/browse/SAGE-655)

### CODE REVIEW

File: packages/sage-react/lib/themes/legacy/Alert/Alert.jsx [hunk 1]
```diff
@@ -13,9 +13,10 @@ export const Alert = ({
   description,
   dismissable,
   icon,
+  onClickDismiss,
```
[the remark this answers is not in the document]
Comment (ju-Skinner) [reply packages/sage-react/lib/themes/legacy/Alert/Alert.jsx:16]: We need test coverage for this as well. I can pair on this if you are uncomfortable with the `testing-library`.

**Review by ju-Skinner — CHANGES_REQUESTED** [review]
See comments.

File: packages/sage-react/lib/themes/legacy/Alert/Alert.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (philschanely) [reply packages/sage-react/lib/themes/legacy/Alert/Alert.jsx:16]: Thanks; changed.

FFR can you possibly provide a link to such conventions documented? Would love for us to have them on hand so we can all follow a common pattern.

File: packages/sage-react/lib/themes/next/Alert/Alert.spec.jsx [hunk 2]
```diff
@@ -70,4 +93,22 @@ describe('Sage Alert', () => {
 
     expect(screen.queryByRole('button')).toBeFalsy();
   });
+
+  it('allows for a callback when the alert is dismissed', async () => {
```
Comment (philschanely) [inline packages/sage-react/lib/themes/next/Alert/Alert.spec.jsx:97]: @ju-Skinner thanks again for the new testing library. It was super easy to use. Welcome any additional critique of what I've added.

**Review by ju-Skinner — APPROVED** [review]

**Pull request merged**